event_model/schemas/event_descriptor.json

```diff
@@ -27,6 +27,13 @@
                         },
                         "description": "The shape of the data.  Null and empty list mean scalar data."
                     },
+                    "dims": {
+                        "type": "array",
+                        "items": {
+                            "type": "string"
+                        },
+                        "description": "The names for dimensions of the data. Null or empty list if scalar data"
```

Comment:
Is there a way to express in the schema that if this is non-null this must be the same length as shape?

Reply:
I could not find one. We can enforce it in the composing tools at least.